### PR TITLE
lib: abstract out sandbox for platforms

### PR DESCRIPTION
**Pull request by vbatts**
Signed-off-by: Vincent Batts <vbatts@hashbangbash.com>

**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM
/test all

**Comment by wking** [comment]
> i can't see what the failure is. No logs available?

You have to drill down a bit.  From [here][1], click “job in Jenkins” to get [here][2].  Under “Identified problems”, click “Indication 1” to get [the error][3].  Scrolling back up a bit from there, the error seems to be:

```
TASK [run integration tests] ***************************************************
task path: /go/src/github.com/kubernetes-incubator/cri-o/contrib/test/integration/test.yml:20
fatal: [localhost]: FAILED! => {
    "changed": false
}

MSG:
---
async task did not complete within the requested time
---
	to retry, use: --limit @/go/src/github.com/kubernetes-incubator/cri-o/contrib/test/integration/main.retry
```

That seems like an underprovisioned-worker flake to me (and we have a few issues for that sort of thing, e.g. #1285).

[1]: https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/kubernetes-incubator_cri-o/1384/test_pull_request_crio_integration_rhel/535/
[2]: https://ci.openshift.redhat.com/jenkins/job/test_pull_request_crio_integration_rhel/535/
[3]: https://ci.openshift.redhat.com/jenkins/job/test_pull_request_crio_integration_rhel/535/consoleFull#53869679658b6e51eb7608a5981914356

File: lib/container_server.go [hunk 1]
```diff
@@ -701,7 +700,7 @@ func (c *ContainerServer) AddSandbox(sb *sandbox.Sandbox) {
 	c.state.sandboxes.Add(sb.ID(), sb)
 
 	c.stateLock.Lock()
-	c.state.processLevels[selinux.NewContext(sb.ProcessLabel())["level"]]++
+	c.addSandboxPlatform(sb)
```
Comment (rhatdan) [inline lib/container_server.go]: Not crazy about the method name.  The method is label or SELinux specific.  Why not c.allocateSecurityLabel(sb), c.freeSecurtyLabel(sb)?

Comment (vbatts) [reply lib/container_server.go]: I *figured* that there may be platform dependent setup needed. This allow an abstraction point. Currently on linux that is only for selinux

Comment (rhatdan) [reply lib/container_server.go]: Ok.  More of a Nit than anything else.

**Comment by rhatdan** [comment]
@runcom @mrunalp PTAL  
Need one more LGTM and we can merge.

**Comment by rhatdan** [comment]
@giuseppe PTAL

**Comment by giuseppe** [comment]
LGTM

**Pull request merged**